xapian-core/weight/tfidfweight.cc

```diff
@@ -70,26 +92,41 @@ TfIdfWeight::name() const
 string
 TfIdfWeight::serialise() const
 {
-    return normalizations;
+    string result = serialise_double(param_slope);
+    result += serialise_double(param_delta);
+    result += normalizations;
+    return result;
 }
 
 TfIdfWeight *
 TfIdfWeight::unserialise(const string & s) const
 {
-    if (s.length() != 3)
+    const char *ptr = s.data();
+    const char *end = ptr + s.size();
+    double slope = unserialise_double(&ptr, end);
+    double delta = unserialise_double(&ptr, end);
+    string normals(ptr, end);
+    ptr += 3;
+    if (rare(ptr != end))
 	throw Xapian::SerialisationError("Extra data in TfIdfWeight::unserialise()");
-    return new TfIdfWeight(s);
+    return new TfIdfWeight(normals, slope, delta);
 }
 
 double
-TfIdfWeight::get_sumpart(Xapian::termcount wdf, Xapian::termcount,
+TfIdfWeight::get_sumpart(Xapian::termcount wdf, Xapian::termcount doclen,
 			 Xapian::termcount) const
 {
     Xapian::doccount termfreq = 1;
     if (normalizations[1] != 'n') termfreq = get_termfreq();
     double wt = get_wdfn(wdf, normalizations[0]) *
 		get_idfn(termfreq, normalizations[1]);
-    return get_wtn(wt, normalizations[2]) * factor;
+    if (normalizations[2] == 'P' && normalizations[1] == 'P') {
+	double wqf = get_wqf();
```

Comment:
Why is this conditional of the second and third normalisations both being `P`?  The documentation you wrote says that xPx and xxP can both be used, but in those terms here you check for xPP .